packages/mail-client/src/mail-client.ts

```diff
@@ -76,6 +76,12 @@ function getCredentials(
   destination: Destination,
   originalProperties: Record<string, any>
 ): { username: string; password: string } {
+  if (
+    destination.proxyType === 'OnPremise' &&
+    destination.authentication === 'NoAuthentication'
+  ) {
+    return { username: '', password: '' };
```

Comment:
Not sure if this is the best way to set empty username and password. According to https://nodemailer.com/smtp/#authentication, if authentication not needed, maybe we should leave `auth` out in `createTransport`.